test/PolledChannelTest.cpp

```diff
@@ -0,0 +1,159 @@
+}
+
+TEST_F(PolledChannelTest, PolledChannelPollPacketToProcessWithReleaseTest) {
+  int channel_capacity = 2;
+  pipert::PolledChannel<int> polled_channel = scheduler_.CreatePolledChannel<int>("TestChannel", channel_capacity);
+  pipert::PacketToFill<int> packet_to_fill = polled_channel.Acquire("TestChannel", pipert::Timer::time(), 42);
+  packet_to_fill.Push();
+  pipert::PacketToProcess<int> packet_to_process = polled_channel.Poll();
+  EXPECT_EQ(polled_channel.GetFreePacketSize(), channel_capacity - 1);
+  EXPECT_EQ(polled_channel.GetQueuedPacketSize(), 0);
+  packet_to_process.Release();
+  EXPECT_TRUE(packet_to_process.IsEmpty());
+  EXPECT_EQ(polled_channel.GetFreePacketSize(), channel_capacity);
+  EXPECT_EQ(polled_channel.GetQueuedPacketSize(), 0);
+}
+
+TEST_F(PolledChannelTest, PolledChannelPollPacketToProcessWithAutoReleaseTest) {
+  int channel_capacity = 2;
+  pipert::PolledChannel<int> polled_channel = scheduler_.CreatePolledChannel<int>("TestChannel", channel_capacity);
+  pipert::PacketToFill<int> packet_to_fill = polled_channel.Acquire("TestChannel", pipert::Timer::time(), 42);
+  packet_to_fill.Push();
+  {
+    pipert::PacketToProcess<int> packet_to_process = polled_channel.Poll();
+    EXPECT_EQ(polled_channel.GetFreePacketSize(), channel_capacity - 1);
+    EXPECT_EQ(polled_channel.GetQueuedPacketSize(), 0);
+  }
+  EXPECT_EQ(polled_channel.GetFreePacketSize(), channel_capacity);
+  EXPECT_EQ(polled_channel.GetQueuedPacketSize(), 0);
+}
+
+TEST_F(PolledChannelTest, PolledChannelDropPacketAfterPushTest) {
+  int channel_capacity = 1;
+  pipert::PolledChannel<int> polled_channel = scheduler_.CreatePolledChannel<int>("TestChannel", channel_capacity);
+  pipert::Timer::Time old_time = pipert::Timer::time();
+  pipert::Timer::Time new_time = pipert::Timer::time();
+  pipert::PacketToFill<int> packet_to_fill = polled_channel.Acquire("TestChannel", old_time, 32);
+  packet_to_fill.Push();
+  packet_to_fill = polled_channel.Acquire("TestChannel", new_time, 42);
+  packet_to_fill.Push();
+  pipert::PacketToProcess<int> packet_to_process = polled_channel.Poll();
+  EXPECT_FALSE(packet_to_process.IsEmpty());
+  EXPECT_EQ(packet_to_process.data(), 42);
+  EXPECT_EQ(packet_to_process.timestamp(), new_time);
+  packet_to_process.Release();
+  EXPECT_EQ(polled_channel.GetFreePacketSize(), channel_capacity);
+  EXPECT_EQ(polled_channel.GetQueuedPacketSize(), 0);
+  EXPECT_TRUE(packet_to_process.IsEmpty());
+}
+
+TEST_F(PolledChannelTest, PolledChannelDropPacketBeforePushTest) {
+  int channel_capacity = 1;
+  pipert::PolledChannel<int> polled_channel = scheduler_.CreatePolledChannel<int>("TestChannel", channel_capacity);
+  pipert::Timer::Time old_time = pipert::Timer::time();
+  pipert::Timer::Time new_time = pipert::Timer::time();
+  pipert::PacketToFill<int> packet_to_fill_old = polled_channel.Acquire("TestChannel", old_time, 32);
+  pipert::PacketToFill<int> packet_to_fill_new = polled_channel.Acquire("TestChannel", new_time, 42);
```

Comment:
So it fails here and it is a bug. It can happen that no droppable job exists and it should still survive